File: 02_Regression/01_Simple Linear Regression/simple_linear_regression.py
# Data Preprocessing Template

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

#Importing the dataset
no_vars = 1
dataset = pd.read_csv('Salary_Data.csv')
x = dataset.iloc[:, :-1].values
y = dataset.iloc[:, no_vars].values


# Feature scaling
"""from sklearn.preprocessing import StandardScaler
sc_x = StandardScaler()
x[:, 3:] = sc_x.fit_transform(x[:, 1:2])"""

# Splitting data into taining and test set
from sklearn.cross_validation import train_test_split
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size = 1/3, random_state = 0)

#Fitting Linear regression to the training set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(x_train, y_train)

#Predicting test set result
y_pred = regressor.predict(x_test)

#Visualizing the training results
plt.scatter(x_train, y_train, color ='red')
plt.plot(x_train, regressor.predict(x_train), color = 'green')
plt.show()
plt.scatter(x_test, y_test, color = 'red')
plt.plot(x_test, y_pred, color = 'blue')

File: 02_Regression/01_Simple Linear Regression/simple_linear_regression.R
# Data Preprocessing Template

# Importing the dataset
dataset = read.csv('Salary_Data.csv')
#dataset = dataset[, 2:3]

# Feature Scaling
#dataset[, 2:3] = scale(dataset[, 2:3])

# Splitting dataset
#install.packages('caTools')
library(caTools)
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

#Fitting Simple linear regression
regressor = lm(formula = Salary ~ YearsExperience, training_set)

#Predicting test results
y_pred = predict.lm(regressor, newdata = test_set)

#Visualizing training results
library(ggplot2)
ggplot() +
  geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary),
             colour = 'red') + 
  geom_line(aes(x = training_set$YearsExperience, predict(regressor, newdata = training_set)),
            colour = 'blue')